File: java/3_constructors-destructors/LiveMusicEvent.java
import java.time.LocalTime;

public class LiveMusicEvent implements AutoCloseable {
    private String artist;
    private LocalTime startTime;
    private int reservedSeats;

    // Parameterized constructor
    public LiveMusicEvent(String artist, LocalTime startTime, int seats) {
        this.artist = artist;
        this.startTime = startTime;
        this.reservedSeats = seats;
        System.out.println("Event constructed: " + artist + " at " + startTime);
        // Other tasks: "Setup stage lights"
    }

    // Default constructor
    public LiveMusicEvent() {
        this("Open Mic Night", LocalTime.of(19, 0), 50);
    }

    @Override
    public void close() { // Renamed from endEvent to satisfy AutoCloseable
        System.out.println("Event ended: Clearing stage for " + artist);
        // Release resources
    }
}

File: python/3_constructors-destructors/live_music_event.py
# live_music_event.py
from datetime import datetime

class LiveMusicEvent:
    def __init__(self, artist: str, hour: int):  # Constructor
        self.artist = artist
        self.start_time = datetime.now()
        self.start_time = self.start_time.replace(hour=hour, minute=0, second=0)
        print(f"Event constructed: {artist} at {self.start_time.strftime('%H:%M')}")
        # Other tasks: prepare_stage()

    def __del__(self):  # Destructor
        print(f"Destructor called: Clearing stage after {self.artist}'s performance")
        # Cleanup: log attendance, release seats

class PastryBatch:
    def __init__(self, type_: str, quantity: int):
        self.type = type_
        self.quantity = quantity
        print(f"Fresh batch constructed: {quantity} {type_} pastries ready!")

    def __del__(self):
        print(f"Batch expired: Cleaning up remaining {self.type} pastries")

# Demo
event = LiveMusicEvent("Jazz Trio", 20)
batch = PastryBatch("Croissants", 50)
del event  # Force destructor (for demo)
# Normally at end of scope or GC